ballerina/modules/wssec/ws_security_methods.bal

```diff
@@ -0,0 +1,204 @@
+public function applyTimestampToken(xml envelope, *TimestampTokenConfig timestampToken) returns xml|Error {
+    if timestampToken.timeToLive <= 0 {
+        return error Error("Invalid value for `timeToLive`");
+    }
+    Document document = check new (envelope);
+    WSSecurityHeader wsSecurityHeader = check addSecurityHeader(document);
+    WsSecurity wsSecurity = new;
+    string securedEnvelope = check wsSecurity.applyTimestampPolicy(wsSecurityHeader, timestampToken.timeToLive);
+    return convertStringToXml(securedEnvelope);
+}
+
+# Apply username token security policy to the SOAP envelope.
+#
+# + envelope - The SOAP envelope
+# + usernameToken - The `UsernameTokenConfig` record with the required parameters
+# + return - A `xml` type of SOAP envelope if the security binding is successfully added or else `wssec:Error`
+public function applyUsernameToken(xml envelope, *UsernameTokenConfig usernameToken) returns xml|Error {
+    Document document = check new (envelope);
+    WSSecurityHeader wsSecurityHeader = check addSecurityHeader(document);
+    WsSecurity wsSecurity = new;
+    string securedEnvelope = check wsSecurity
+        .applyUsernameTokenPolicy(wsSecurityHeader, usernameToken.username, usernameToken.password,
+                                  usernameToken.passwordType);
+    return convertStringToXml(securedEnvelope);
+}
+
+# Apply symmetric binding security policy with username token to the SOAP envelope.
+#
+# + envelope - The SOAP envelope
+# + symmetricBinding - The `SymmetricBindingConfig` record with the required parameters
+# + return - A `xml` type of SOAP envelope if the security binding is successfully added or else `wssec:Error`
+public function applySymmetricBinding(xml envelope, *SymmetricBindingConfig symmetricBinding) returns xml|Error {
+    Document document = check new (envelope);
+    WSSecurityHeader wsSecurityHeader = check addSecurityHeader(document);
+    string securedEnvelope = envelope.toBalString();
+    if symmetricBinding.signatureAlgorithm !is () {
+        Signature signature = check new ();
+        byte[] signedData = check signature.signData((envelope/<soap:Body>/*).toString(),
+                                                     <SignatureAlgorithm>symmetricBinding.signatureAlgorithm,
+                                                     symmetricBinding.symmetricKey);
+        Signature signatureResult = check addSignature(signature,
+                                                       <SignatureAlgorithm>symmetricBinding.signatureAlgorithm,
+                                                       signedData);
+        WsSecurity wsSecurity = new;
+        securedEnvelope = check wsSecurity.applySignatureOnlyPolicy(wsSecurityHeader, signatureResult,
+                                                                    symmetricBinding.x509Token);
+    }
+    if symmetricBinding.encryptionAlgorithm !is () {
```

Comment:
```suggestion
    if symmetricBinding.encryptionAlgorithm is EncryptionAlgorithm {
```

Reply:
Fixed in [1de6750](https://github.com/ballerina-platform/module-ballerina-soap/pull/103/commits/1de6750f3a32d73e90b0ade50476774f2541fccf)